### DISCUSSION TITLE
Migrations Subfolders

### DISCUSSION THREAD

**Discussion by yusufalper** [laravel/framework#47521] [Ideas]
See the [Pull Request](https://github.com/laravel/framework/pull/47527)

By default, Laravel expects all migration files to be placed directly in the migrations directory without any subdirectories. 
I have written a [Service Provider](https://gist.github.com/yusufalper/25c0c7e5a347abc09349bbc16817c268) for this and I use it in almost every project of mine. When I researched it, I saw many people doing the same thing, and I don't think it would cause any harm. Can this feature be added to the framework? If the answer is negative, why shouldn't it be? In some cases, it can be really necessary because instead of creating new migrations, updates and changes are usually made on existing ones until the project reaches a certain stage, and this feature would be useful for categorizing migrations. By the way, there is more description in the gist link.

**Comment by shaedrich** [comment]
I'm curious as to what the use cases were:
* a folder for each model
* a folder for action (create, alter, delete)
* a folder per module (DDD)
* a folder per major version

Is that what you envisioned? Are there other use cases?

**Reply by yusufalper** [reply]
I can provide an example, and there can be many examples depending on individual preferences. It's not something that will cause harm. Here's my example: I'm working on a project with multiple databases (6 in total), and I'm not working on a module-based structure. There are approximately 200 migration files. I want to organize them under a structure; first based on the database and then based on functionality within the 'migrations' folder. But It has already been rejected, and I don't think there's any point in continuing the discussion. It's an action that can already be done with a simple ServiceProvider anyway..

**Reply by shaedrich** [reply]
Thanks for the explanation. I was just curious because I did something similar. It's always so many migration files when they are all in the same folder which quickly gets cluttered. I went with the second approach to at least group them by action.
